### PR TITLE
Handle multivalue X-Forwarded-Proto header

### LINKED ISSUE #5198: X-Forwarded-Proto overwrites the scheme of HTTP triggers
#### Repro steps
Call a HTTP-triggered function with the HTTP header:
`X-Forwarded-Proto: https,https`

#### Expected behavior
Calls function.

#### Actual behavior
ArgumentException from System.Private.Uri: `Invalid URI: The URI scheme is not valid.`

#### Known workarounds
None that I'm aware of. In my case, I don't have control of the client that is calling my function.

#### Related information 
I walked through the call stack and found the `HttpContext`'s `Request` was having its `Scheme` changed by `AppServiceHeaderFixupMiddleware`. Is there a way to opt out of this?
You can see the [offending code here](
https://github.com/Azure/azure-functions-host/blob/f9540d6aaafc56ecd3d8d045edfc159a85ea9ba1/src/WebJobs.Script.WebHost/Middleware/AppServiceHeaderFixupMiddleware.cs#L32).

I believe it's a valid header. Here's an appeal to authority anyway:

> **Nick Craver @Nick_Craver 10:57 AM · Apr 15, 2018**
> If you see this...
> 
> if (headers["X-Forwarded-Proto"] == "https") { ... }
> 
> ...it's wrong. For example, these are all valid headers:
> X-Forwarded-Proto: https, https
> X-Forwarded-Proto: http, https
> X-Forwarded-Proto: https, http
> It's <client>, <proxy1>, <proxy2>, etc.

https://twitter.com/nick_craver/status/985321287194210304

```
Azure Functions Core Tools (2.7.1846 Commit hash: 458c671341fda1c52bd46e1aa8943cb26e467830)
Function Runtime Version: 2.0.12858.0

<TargetFramework>netcoreapp2.1</TargetFramework>
<AzureFunctionsVersion>v2</AzureFunctionsVersion>
<PackageReference Include="Microsoft.NET.Sdk.Functions" Version="1.0.30-beta2" />
```

### PR DESCRIPTION
**Pull request by ahmelsayed**
Closes #5198

### CODE REVIEW

**Comment by ahmelsayed** [comment]
@brettsam I pushed another change to only register this middleware in linux consumption. It shouldn't be in the CLI or the runtime generically like it was.

File: src/WebJobs.Script.WebHost/Middleware/AppServiceHeaderFixupMiddleware.cs [hunk 1]
```diff
@@ -29,7 +29,7 @@ public async Task Invoke(HttpContext httpContext)
 
             if (httpContext.Request.Headers.TryGetValue(ForwardedProtocolHeader, out value))
             {
-                httpContext.Request.Scheme = value;
+                httpContext.Request.Scheme = value.FirstOrDefault();
```
[the remark this answers is not in the document]
Comment (ahmelsayed) [reply src/WebJobs.Script.WebHost/Middleware/AppServiceHeaderFixupMiddleware.cs]: Good point. I don't think a null scheme, but I think it makes sense to not overwrite it if the header happens to be empty for any reason. Thanks @NickDarvey

File: src/WebJobs.Script.WebHost/Middleware/AppServiceHeaderFixupMiddleware.cs [hunk 2]
```diff
@@ -29,7 +29,11 @@ public async Task Invoke(HttpContext httpContext)
 
             if (httpContext.Request.Headers.TryGetValue(ForwardedProtocolHeader, out value))
             {
-                httpContext.Request.Scheme = value.FirstOrDefault();
+                var scheme = value.FirstOrDefault();
```
Comment (fabiocav) [inline src/WebJobs.Script.WebHost/Middleware/AppServiceHeaderFixupMiddleware.cs]: nit: `var` -> `string`

File: src/WebJobs.Script.WebHost/Middleware/AppServiceHeaderFixupMiddleware.cs [hunk 3]
```diff
@@ -29,7 +29,11 @@ public async Task Invoke(HttpContext httpContext)
...
+                if (scheme != null)
```
Comment (fabiocav) [inline src/WebJobs.Script.WebHost/Middleware/AppServiceHeaderFixupMiddleware.cs]: Could we end up with an empty string here as well?

Comment (ahmelsayed) [reply src/WebJobs.Script.WebHost/Middleware/AppServiceHeaderFixupMiddleware.cs]: as NickDarvey mentioned, an empty header value is a malformed header. TryGetValue returns false if the header is passed as empty from the client, so I was assuming aspnet will do validation on these things before passing the request to the application. `!string.IsNullOrEmpty()` won't hurt though, even if technically not valid per https://tools.ietf.org/html/rfc7230#section-3.2

Comment (fabiocav) [reply src/WebJobs.Script.WebHost/Middleware/AppServiceHeaderFixupMiddleware.cs]: Makes sense, just wanted to make sure it's something we had validated/thought about.

**Pull request merged**